charts/metallb/crds/crds.yaml

```diff
@@ -25,8 +22,8 @@ spec:
         # but we're going to set it later using the cert-controller.
         caBundle: Cg==
         service:
-          namespace: {{ .Release.Namespace }}
-          name: {{ template "metallb.fullname" . }}-webhook-service
+          namespace: default
```

Comment:
Should `namespace` be `default` here?  Will it get patched later?

Reply:
as far as i understood from the docs if we use the crds/ we can't template the file

Reply:
Looks like this is addressed here: https://github.com/helm/helm/issues/8171

Reply:
So if I get the suggestion right, the idea is to provide a separate chart (without crds/, just a regular chart) to deploy the crds, and that can be templated. Is that right @gclawes ?
We'll have to publish two separate charts and adjust our release toolery, but that won't be an issue I guess.

Reply:
from what i saw so far, i think keeping the chart as it is would be the best. i don't like the fact the crds are not templated and i don't see the advantage of adding another chart just for crds, but maybe i'm missing something here.

Reply:
I think it may be possible to use a sub-chart to do this: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/

This would probably solve our issues around templating the webhook config in the CRDs, though it does mean the CRDs get auto-updated whenever they are changes (unless operations want to use `crds.enabled` to toggle them, see below).

I know some projects like `kube-prometheus-stack` are very deliberate about updating breaking CRD changes outside of helm (see [here](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#upgrading-chart)), but they also a) use the `crds/` directory and b) do not use upgrade webhooks.  That being said, I'm _fairly_ sure the subchart approach would work for us.

Assuming we'd name the CRD sub-chart chart `crds`, place it in `charts/metallb/charts/crds/`.  Template values for that could be put in `charts/metallb/values.yaml` under `crds:`

In `charts/metallb/Chart.yaml`, add this as a dependency to allow the CRD sub-chart to be toggled (otherwise it is just enabled by default):
```
dependencies:
- name: crds
  condition: "crdsenabled"
```

This way, the sub-chart will just be packaged with the main chart (see my example of `main-chart`/`sub-chart` below).  I think the only change to the release process would be to update `version:`/`appVersion:` in `charts/metallb/charts/crds/Chart.yaml`